### PR TITLE
ISPN-7073 XSD Documentation improvements

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-7073

ISPN-7073 Update XSDoc tool
    - support complexTypes with choice children
    - update the Google Analytics tracking code
    - generate docs only for the latest version of each namespace
    - output using html format instead of XML to avoid corruption of empty tags

Also
- Remove unused server schemas
- Generate the html reference for server schemas

### CODE REVIEW

**Comment by tristantarrant** [comment]
If you want to test this:
```
mvn clean install -DskipTests
mvn install -Pdistribution distribution
```
Then check in distribution/target/site/configdocs and distribution/target/site/serverconfigdocs

File: distribution/pom.xml [hunk 1]
```diff
@@ -11,22 +11,6 @@
    <name>Infinispan Distribution</name>
    <packaging>pom</packaging>
 
-   <properties>
-      <analytics.code>
----&gt;
--&lt;!--
--&lt;script type='text/javascript'&gt;
--var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
--document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
```
Comment (slaskawi) [inline distribution/pom.xml]: The analytics code has been refactored [in this PR](https://github.com/infinispan/infinispan/pull/4440). I would prefer if you didn't touch it and integrate my PR instead!

**Comment by tristantarrant** [comment]
I've rebased this after the merge of the javadoc PR

**Comment by slaskawi** [comment]
Looking...

**Comment by slaskawi** [comment]
Looks good, let's try the new "rebase and merge button" :D

**Comment by slaskawi** [comment]
Integrated, thanks Tristan!

**Pull request merged**